samples: net: sockets: Update docs with TLS information

rlubos opened the pull request:
Update documentation of socket samples that support TLS with information
on how to enable it.

Partially fixes #9926

Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>

rlubos committed:
samples: net: sockets: Update docs with TLS information

Update documentation of socket samples that support TLS with information
on how to enable it.

Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>

dbkinder commented on samples/net/sockets/http_get/README.rst:
```diff
@@ -45,6 +45,31 @@ and dumps the response. You can edit the source code to issue a request
+``overlay-tls.conf`` overlay config file enabled, for example:
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/net/
```
Probably:

    :zephyr-app: samples/net/sockets/http_get

dbkinder commented on samples/net/sockets/http_get/README.rst:
```diff
@@ -45,6 +45,31 @@ and dumps the response. You can edit the source code to issue a request
+An alternative way is to specify ``-DOVERLAY_CONFIG=overlay-tls.conf`` when
+running cmake.
+
+Certificate used by the sample can be found in the ``src`` directory. The
```
Change to:

    The certificate used by the sample can be found in the sample's ``src`` directory.

dbkinder commented on samples/net/sockets/http_get/README.rst:
```diff
@@ -45,6 +45,31 @@ and dumps the response. You can edit the source code to issue a request
+
+Certificate used by the sample can be found in the ``src`` directory. The
+certificate was selected to enable access to the default website configured in
+the sample (https://google.com). In order to access different webpage over TLS
```
Change to:

    To access a different web page over TLS, provide an appropriate certificate to
    authenticate to that web server.

rlubos commented:
@dbkinder Thank you for the feedback, I've addressed your comments.

I had a doubt about one comment, please check if slightly modified proposal is fine.

dbkinder reviewed (approved):
doc changes look good, thanks.

rlubos commented:
recheck
